manual/embedded-examples/src/docs/dev/procedures.asciidoc

```diff
@@ -0,0 +1,495 @@
+       <executions>
+         <execution>
+           <phase>package</phase>
+           <goals>
+             <goal>shade</goal>
+           </goals>
+         </execution>
+       </executions>
+     </plugin>
+   </plugins>
+  </build>
+----
+
+Until the GA release of Neo4j 3.0, the dependency on Neo4j requires that a _snapshot repository_ is configured.
+This repository is where Maven will find the latest build of Neo4j to use as a dependency.
+
+[source, xml]
+----
+  <repositories>
+   <repository>
+     <id>neo4j-snapshot-repository</id>
+     <name>Maven 2 snapshot repository for Neo4j</name>
+     <url>http://m2.neo4j.org/content/repositories/snapshots</url>
+     <snapshots><enabled>true</enabled></snapshots>
+     <releases><enabled>false</enabled></releases>
+   </repository>
+ </repositories>
+</project>
+----
+
+
+=== Writing integration tests
+
+In the test dependencies are included Neo4j Harness and JUnit.
+These can be used to write integration tests for the procedure.
+
+First we decide what the procedure should do, then we write a test that proves that it does it right.
+Finally we write a procedure that passes the test.
+
+Below is a template for testing a procedure that accesses Neo4j full-text indexes from Cypher.
+
+.Writing tests for procedures
+[source, java]
+----
+package example;
+
+import org.junit.Rule;
+import org.junit.Test;
+
+import org.neo4j.harness.junit.Neo4jRule;
+
+import static org.neo4j.bolt.BoltKernelExtension.Settings.connector;
+import static org.neo4j.bolt.BoltKernelExtension.Settings.enabled;
+
+public class LegacyFullTextIndexTest
+{
+   // This rule starts a Neo4j instance for us
+   @Rule
+   public Neo4jRule neo4j = new Neo4jRule()
+
+           // This is the Procedure we want to test
+           .withProcedure( FullTextIndex.class )
+
+           // Temporary until Neo4jRule includes Bolt by default
+           .withConfig( connector( 0, enabled ), "true" );
+
+   @Test
+   public void shouldXYZ() throws Throwable
+   {
+       // Write your test code here, for instance using
+       // the Neo4j Java Driver
+   }
+}
+----
+
+
+[[writing-procedure]]
+=== Writing a procedure
+
+With the test in place, we write a procedure.
+The full example is available in the {procedure-template-url}[Neo4j Procedure Template] repository.
+
+Particular things to note:
+
+* All procedures are annotated `@Procedure`.
+  Procedures that write to the database are additionally annotated `@PerformsWrites`.
+* The _context_ of the procedure, which is the same as each resource that the procedure wants to use, is annotated `@Context`.
+* The _input_ and _output_.
+
+For more details, see the link:javadocs/index.html?org/neo4j/procedure/Procedure.html[API documentation for procedures].
+
+// .Todo
+// --
+// * [ ] Input types
+// * [ ] How to declare output
+// * [ ] Injectable resources
+// * [ ] Say that it's a _method_, not a _class_.
+// --
+
+
+[source, java]
+----
+package example;
```

Comment:
should that file life in the manual repo? currently there is a public copy in github.com/neo4j-examples